src/core/ext/lb_policy/grpclb/grpclb.c

```diff
@@ -1007,8 +1011,10 @@ static void lb_client_destroy(glb_lb_policy *glb_policy) {
 static void query_for_backends_locked(grpc_exec_ctx *exec_ctx,
                                       glb_lb_policy *glb_policy) {
   GPR_ASSERT(glb_policy->lb_channel != NULL);
+  /* take a weak ref (won't prevent calling of \a glb_shutdown if the strong ref
+   * count goes to zero) to be unref'd in lb_on_server_status_received */
   GRPC_LB_POLICY_WEAK_REF(&glb_policy->base, "query_for_backends_locked");
```

Comment:
Now that I think about it, don't we actually need to take two weak refs here, one for each of the pending callbacks we're about to create (`lb_on_server_status_received` and `lb_on_response_received`)?  If we get shut down, they will probably both be in flight at the same time, and we'll need to handle both of them.